### ISSUE TITLE
Split source in editor

### ISSUE THREAD

**Issue by mine-cetinkaya-rundel** [rstudio/rstudio#2129]
In TexShop there’s an option to split source which allows you to see (and edit) multiple areas in a document in one window so that you don’t need to keep scrolling up and down. It would be wonderful to have this as a functionality in the RStudio IDE editor, especially for R Markdown documents. See the image below for what it looks like in TexShop.

<img width="602" alt="screen shot 2018-02-04 at 12 17 25 pm" src="https://user-images.githubusercontent.com/5965649/35782100-f07875f4-09a7-11e8-8733-2c46cab59de8.png">

**Comment by kevinushey** [comment]
I'm going to optimistically place this into v1.2 as I think it would be awesome to make this part of the next release, but depending on time constraints we may need to push this back.

**Comment by Avi-Kenny** [comment]
Not sure if there is a place for users to "vote" to prioritize new features, so I'm doing so here. This feature would be huge for my workflow (and I know many others' as well).

**Comment by altamont** [comment]
This feature would be massively appreciated.  Especially when editing a shiny server file that is >1000 lines long.

**Comment by arne-henningsen** [comment]
Yes, this feature would be very useful in many circumstances. Are there any plans for implement this feature?

**Comment by kwongyuw** [comment]
Splitting source view would be very helpful for long script file.

**Comment by ashenkin** [comment]
Thanks @jmcphers.  Agree with this issue.  Just want to chip in that splitting source in a single window isn't the only way to do it.  I'd personally like to be able to have multiple windows open to the same file at the same time.

**Comment by jmcphers** [comment]
@ashenkin Yes, that's true! I'll update the title.

**Comment by sbalci** [comment]
I vote for this feature too :)

**Comment by klarchristian** [comment]
agreed - it would be very helpful.

**Comment by IFFranciscoME** [comment]
This feature will be very helpful and a major tool to open even 2 different files in the same window (verically splitted). Because, a number of years ago, RStudio was mainly for R, but nowadays it supports several other languages and the mix of those in multilanguage projects is critical for today's data science workflow. So yeah, do please add the functionality for vertical split the, if you like, two different files as a starter.

**Comment by iago-pssjd** [comment]
Currently not only TextShop, but other softwares like Konqueror and Visual Studio Code have this feature which allows to split the window vertically or horizontally with the same or distinct files. I agree it will be a major feature.

**Comment by jmcphers** [comment]
Recently requested on the community forum: https://community.rstudio.com/t/cant-seem-to-open-the-same-file-in-two-code-columns/94681

**Comment by leowill01** [comment]
This would be greatly appreciated. Right now my only workaround is to open the same source file in VS Code and view different parts of the file there while I maintain view of the section I'm working on in RStudio.

**Comment by mmahmoudian** [comment]
This feature has been asked many times in many forms (another example below) and I really thing this feature is one of the major feature that lacks currently in Rstudio

https://community.rstudio.com/t/split-panes-in-source-editor/11992

I wonder what can be done to boost the process (of course other than developing it myself and create a pull request).

**Comment by tkasci** [comment]
I agree that this would be an extremely useful feature for complex files and especially Shiny app programming.

**Comment by mtcruz** [comment]
I also don't know if this is the place to upvote the implementation of new features, but I'm doing so here.

**Comment by wkumler** [comment]
Just wanting to give some continued support to this feature! Looks like it's still on your radar but I'm very excited for it.

**Comment by mikebessuille** [comment]
Kevin did a proof of concept with https://github.com/rstudio/rstudio/pull/486

**Comment by rvanmazijk** [comment]
Adding my support for this much-needed feature! Especially useful for long RMarkdown documents with lots of figures, where I often like to have a figure-rich section open for reference while I'm writing elsewhere in the document.

**Comment by jucor** [comment]
Any chance to have this, please? Extremely helpful when writing long Rmarkdown files and needing to refer to previous part.

**Comment by fabp5** [comment]
Agreed - this would be very helpful, especially for long Rmarkdown and Shiny files. I often end up copying and pasting my code into a text editor too so I can view two parts at once, which isn't ideal.

**Comment by JonasEngstrom** [comment]
I would also appreciate this feature.

**Comment by avhz** [comment]
Any progress on this feature ?
